src/presenter/point-presenter.js

```diff
@@ -0,0 +1,112 @@
+import { render, replace } from '../framework/render.js';
+import EventsView from '../view/event-view.js';
+import EventEdit from '../view/event-edit.js';
+
+const PointMode = {
+  VIEW: 'view',
+  EDIT: 'edit',
+};
+
+export default class PointPresenter{
+
+  #eventsListView = null;
+  #point = null;
+  #offer = null;
+  #destination = null;
+  #points = null;
+
+  #pointComponent = null;
+  #pointEditComponent = null;
+  #openEditModeHandler = null;
+
+  #mode = null;
+  #modeFavorit = null;
+  #openFavoritModeHandler = null;
+  #changeFavorite = null;
+
+  constructor({eventsListView, point, offers, destinations}){
+    this.#eventsListView = eventsListView;
+    this.#point = point;
+    this.#offer = offers;
+    this.#destination = destinations;
+    this.#mode = PointMode.VIEW;
+    this.#modeFavorit = this.#point.isFavorite;
+  }
+
```

Comment:
Я бы советовал разделить `renderPoint` на 5 методов, 2 на создание каждого вида вьюшки: отображение/редактирование
```suggestion
 #createViewModeComponent(point) {
    const pointView = new EventsView({
      point,
      pointDestination: this.#destinationsModel.getById(point.destination),
      pointOffers: this.#offersModel.getByType(point.type)
    });

    pointView.setEditHandler(this.openEditMode);
    pointView.setFavoriteHandler(this.changeFavorite);

    return pointView;
  }
  
   #createEditModeComponent(point) {
    const pointEdit = new EventEdit({
      point,
      pointDestination: this.#destinationsModel.get(),
      pointOffers: this.#offersModel.get()
    });

    pointEdit.setEditHandler(this.closeEditMode);

    return pointEdit;
  }
```

Reply:
Два на отрисовку каждого из этих виднов: один отрисовывает редактирование, второй просмотр
```suggestion
 #renderEditMode() {
    const view = this.#createEditModeComponent(this.#pointData);
    this.#renderOrReplace(view);

    this.#pointEditComponent = view;
    this.#mode = PointMode.EDIT;
  };

  #renderViewMode() {
    const view = this.#createViewModeComponent(this.#pointData);
    this.#renderOrReplace(view);

    this.#pointViewComponent = view;
    this.#mode = PointMode.VIEW;
  };
```